crates/adkg/src/adkg.rs

```diff
@@ -993,7 +983,24 @@ where
                     }
                 }
             }
-            final_parties.into_iter().collect()
+
+            let mut final_sessions = HashSet::with_capacity(state.n);
+            for j in aba_sessions {
+                // Get the list of parties for that session
+                let rbc_sessions = match state.rbc_outputs.get(&j) {
+                    Some(rbc_sessions) => rbc_sessions,
+                    None => {
+                        info!(
+                            "Node `{}` obtained an estimate from ABA with sid `{j}` without having obtained an RBC output, waiting.",
+                            state.id
+                        );
+                        Self::wait_for_rbc_output(&state.rbc_outputs, &j).await
```

Comment:
Could this be triggered by a malicious party remotely and cause deadlock / DoS ?